dspace-swordv2/src/main/java/org/dspace/sword2/ContainerManagerDSpace.java

```diff
@@ -824,22 +781,22 @@ protected void doContainerDelete(SwordContext swordContext, Item item, AuthCrede
 			if (wft.isItemInWorkspace(swordContext.getContext(), item))
 			{
 				WorkspaceItem wsi = wft.getWorkspaceItem(context, item);
-				wsi.deleteAll();
+				workspaceItemService.deleteAll(context, wsi);
 			}
 			else if (wft.isItemInWorkflow(context, item))
 			{
 				InProgressSubmission wfi = wft.getWorkflowItem(context, item);
-				wfi.deleteWrapper();
+				workflowItemService.deleteWrapper(context, wfi);
 			}
 
 			// then delete the item
-			Collection[] collections = item.getCollections();
+			List<Collection> collections = item.getCollections();
 			for (Collection collection : collections)
```

Comment:
This will probably give a concurrentModificationException, best to use itemService.delete(context, item);